static/js/message_list_data.js

```diff
@@ -42,6 +42,21 @@ export class MessageListData {
         return this._items.at(-1);
     }
 
+    ids_greater_or_equal_than(my_id) {
+        const result = [];
+
+        for (let i = this._items.length - 1; i >= 0; i -= 1) {
+            const message_id = this._items[i].id;
+            if (message_id >= my_id) {
+                result.push(message_id);
+            } else {
+                break;
```

Comment:
I changed this `break` to a `continue`; there's no real performance impact either way and it relies less on the IDs being sorted.